diff --git a/node_modules/@web/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js b/node_modules/@web/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js
index 1d0280c..f4aeb21 100644
--- a/node_modules/@web/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js
+++ b/node_modules/@web/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js
@@ -58,10 +58,14 @@ function importMetaAssets({ include, exclude, warnOnError, transform } = {}) {
     name: 'rollup-plugin-import-meta-assets',
 
     async transform(code, id) {
+      console.log({ id });
       if (!filter(id)) {
+        console.log('NULL @@@@@', { code });
         return null;
       }
 
+      console.log('here!?!!?!?!');
+
       const ast = this.parse(code);
       const magicString = new MagicString(code);
       let modifiedCode = false;
